lib/modules/gasEstimator/index.js

```diff
@@ -0,0 +1,72 @@
+const async = require('async');
+const _ = require('underscore');
+const ContractFuzzer = require('../fuzzer');
+
+/*global web3*/
+
+class GasEstimator {
+  constructor(embark) {
+    this.embark = embark;
+    this.logger = embark.logger;
+    this.events = embark.events;
+    this.fuzzer = new ContractFuzzer(embark);
+  }
+
+  estimateGas(contractName, cb) {
+    const self = this;
+    let gasMap = {};
+    self.events.request('contracts:contract', contractName, (contract) => {
+      self.logger.info("Generating three rounds of fuzz for contract -- ", contractName);
+      let fuzzMap = self.fuzzer.generateFuzz(3, contract);
+      self.logger.info("--  Beginning gastimation for contract -- " + contractName);
+      let contractObj = new web3.eth.Contract(contract.abiDefinition, contract.deployedAddress);
+      async.each(contract.abiDefinition, 
+        (abiMethod, gasCb) => {
+          let name = abiMethod.name;
+          console.log("NAME: ", name);
+          console.log("ABI INPUTS: ", abiMethod.inputs);
+          if (abiMethod.type === "constructor") {
+            // already provided for us
+            gasCb(null, 'constructor', contract.gasEstimates.creation.totalCost.toString());
+          } else if (abiMethod.type === "fallback") {
+            gasCb(null, 'fallback', contract.gasEstimates.external[""].toString());
```

Comment:
For the two previous callbacks, what are you trying to achieve? Because those parameters to send to them are not used anywhere and you don,t modify the gasMap with it.

Reply:
ah crap you're right.